### PR TITLE
update recipe requirements for 2.10

### PR DESCRIPTION
**Pull request by CagtayFabry**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] ~~Reset the build number to `0` (if the version changed)~~
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

## Changes
This draft updates the 2.10 release to have matching python requirements as found in the python package

### CODE REVIEW

**Comment by CagtayFabry** [comment]
@WilliamJamieson this draft is an example how to update the 2.10 release on conda to match the requirements specified in `setup.cfg` of `asdf`

I expect the CI to fail here since the `asdf-standard` is not available on conda yet

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,14 +23,16 @@ requirements:
...
+    - python >=3.7
     - pyyaml >=3.10
     - jmespath >=0.6.2
     - Jsonschema >=3.0.2,<4
     - numpy >=1.10
     - semantic_version >=2.8
     - importlib_resources >=3
     - packaging >=16.0
+    - asdf-standard >=1.0.0
+    - asdf-transform-schemas >=0.2.0
```
Comment (WilliamJamieson) [inline recipe/meta.yaml:35]:
```suggestion
    - asdf-standard >=1.0.1
    - asdf-transform-schemas >=0.2.2 
```
These are the versions that are or will be available on conda-forge starting out.

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]
With these changes and the merger of #18146, you should take this PR out of draft.

**Review by WilliamJamieson — APPROVED** [review]

**Comment by CagtayFabry** [comment]
thank you @WilliamJamieson I wasn't around for a few days

glad to see this working now 👍

**Pull request merged**